### PR TITLE
Functools partial reconstruct

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117721

Adds a reconstruction for functools partial that cross graph breaks.

cc @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/variables/functions.py [hunk 1]
```diff
@@ -646,6 +646,21 @@ def get_val(v):
                 **{k: get_val(v) for k, v in self.keywords.items()},
             )
 
+    def reconstruct(self, codegen):
+        codegen.load_import_from("functools", "partial")
+        codegen(self.func)
+        if self.args and self.keywords:
+            # TODO(voz): Is there a util to help with this? neither the existing ones suffice?
+            unimplemented("Reconstruction of mixed args and keywords NYI")
```
Comment (voznesenskym) [inline torch/_dynamo/variables/functions.py:654]: @jansel do we need this? godbolt says python unpacks mixed kwargs/args as just args https://godbolt.org/z/5T6eKefrv but I don't know if that's always right...

Comment (jansel) [reply torch/_dynamo/variables/functions.py:654]: https://docs.python.org/3/library/dis.html#opcode-CALL_FUNCTION_EX

**Review by jansel — APPROVED** [review]

**Comment by voznesenskym** [comment]
Updated - added test case for mix args and kwargs. Ty @jansel!

**Comment by jansel** [comment]
Replaced by #118583

**Pull request closed without merging**